src/Network/Ethereum/Api/Provider.hs

```diff
@@ -51,22 +56,27 @@ instance Exception Web3Error
 
 --TODO: Change to `HttpProvider ServerUri | IpcProvider FilePath` to support IPC
 -- | Web3 Provider
-data Provider = HttpProvider String
+data Provider = HttpProvider String | WSProvider String Int
   deriving (Show, Eq, Generic)
 
-instance Default Provider where
```

Comment:
Why default provider couldn't be HttpProvider?

Reply:
It was done to provide default provider for both HTTP and WebSocket.

Reply:
Yes, I see, but Default instance could be stayed for HTTP provider.